src/main/java/com/develhope/spring/order/service/OrderService.java

```diff
@@ -12,9 +16,38 @@
 public class OrderService {
     @Autowired
     private OrderRepository orderRepository;
+    @Autowired
+    private VehicleRepository vehicleRepository;
 
-    public Order createOrder(Order order) {
-        return orderRepository.save(order);
+    public Order shopVehicle(Vehicle vehicle, VehicleState isPurchasable, User userId) {
+        if (vehicle == null || isPurchasable == null) {
+            throw new IllegalArgumentException("Vehicle and isPurchasable parameters cannot be null.");
+        }
+        if (vehicle.getVehicleState() == isPurchasable && vehicle.isAvailable()) {
+            Order newOrder = new Order();
+            newOrder.setVehicle(vehicle);
+            newOrder.setUser(userId);
+            orderRepository.save(newOrder);
+            return newOrder;
+        } else {
+            throw new OrderNotFoundException("Vehicle purchase failed, the vehicle is not purchasable.");
+        }
+    }
+
+    public Order rentVehicle(Vehicle vehicle, VehicleState isRentable, User userId) {
+        if (vehicle == null || isRentable == null) {
+            throw new IllegalArgumentException("Vehicle and isRentable parameters cannot be null.");
+        }
+
+        if (vehicle.getVehicleState() == isRentable && vehicle.isAvailable()) {
+            Order newOrder = new Order();
+            newOrder.setVehicle(vehicle);
+            newOrder.setUser(userId);
+            orderRepository.save(newOrder);
```

Comment:
puoi ritornare direttamente questo

Reply:
ma sbaglio o non setta alcuno stato all'ordine , ci troviamo così poi con ordini che non sappiamo cosa sono , se ordini acquisti o noleggi, in pratica hanno tutti lo stesso codice ....o facciamo un metodo di creazione ordine generico per i veicoli validi e poi lo richiamino in altri metodi per fare le tipologie di ordine specifiche o altrimenti facciamo già quello perché così non capisco come funzioni .

Reply:
Ma non devo settare lo stato, qua controllo che il veicolo sia affittabile e disponibile, a quel punto si che posso creare l'ordine di affitto
Che faccia riferimento ad un determinato user
E ad un determinato veicolo

Reply:
a parte che il tuo controllo su "affittabile" consiste solo che lo stato non sia non disponibile ma ripeto , l'ordine da te creato in tutte queste pull request è identico, non ha senso fare degli end Point in questo modo

Reply:
senza considerare che i rent dovrebbero essere proprio un'entità separata dagli order perchè hanno attributi diversi dagli acquisti e dagli ordini

Reply:
confermo quanto detto, se controlli luchidChart vedrai che rent era già stata pensata come una entity distinta

Reply:
Perdonami la svista Aurora ma Silvio ha ragione, i rent hanno una classe distinta e vanno trattati come tali. Inoltre sono d'accordo sul fatto che quando un'ordine viene effettuato lo status del veicolo (in questo caso boolean rented) e quello dell'ordine stesso vadano aggiornati